infrastructure/src/main/java/org/corfudb/infrastructure/management/PollReport.java

```diff
@@ -83,23 +86,34 @@
     }
 
     /**
-     * All active Layout servers have been sealed but there is no client to take this forward and
-     * fill the slot by proposing a new layout. This is determined by the outOfPhaseEpochNodes map.
+     * Check to see if the current epoch]register has been decided/filled. We do this by
+     * comparing the epoch of the latest commited layout and the epoch associated with the
+     * {@link WrongEpochException}.
      *
-     * @return True if latest layout slot is vacant.
+     * @return Optional.of(epoch) if latest layout slot is vacant, Optional.none() otherwise.
      */
-    public boolean isCurrentLayoutSlotUnFilled() {
-        log.trace("isCurrentLayoutSlotUnFilled. Wrong epochs: {}, active servers: {}",
-                wrongEpochs, responsiveServers
+    public Optional<Long> getLayoutSlotUnFilled(Layout latestCommittedLayout) {
+        log.trace("getLayoutSlotUnFilled. Wrong epochs: {}, latest layout epoch: {}",
+                wrongEpochs, latestCommittedLayout.getEpoch()
         );
 
-        // Check if all active layout servers are present in the outOfPhaseEpochNodes map.
-        boolean result = wrongEpochs.keySet().containsAll(responsiveServers);
-
-        if (result) {
-            log.info("Current layout slot is empty. Filling slot with current layout. " +
-                    "Wrong epochs: {}, active layout servers: {}", wrongEpochs, responsiveServers);
+        long maxOutOfPhaseEpoch = Collections.max(wrongEpochs.values());
+        if (maxOutOfPhaseEpoch > latestCommittedLayout.getEpoch()) {
+            return Optional.of(maxOutOfPhaseEpoch);
         }
-        return result;
+
+        return Optional.empty();
+    }
+
+
+    /**
+     * Given the latest committed layout, figure out if the majority of the layout servers have
+     * been sealed.
+     *
+     * @param latestCommittedLayout latest commited layout
+     * @return true f the majority of the layout servers have been sealed.
+     */
+    public boolean isMajorityLayoutServersSealed(Layout latestCommittedLayout) {
+        return wrongEpochs.size() > latestCommittedLayout.getLayoutServers().size()/2;
```

Comment:
code reformat please

Reply:
Not sure I understand -- what's wrong with it? Also that code has changed.